scripts/azureml-assets/azureml/assets/config.py

```diff
@@ -524,52 +542,62 @@ def __init__(self, storage_name: str, container_name: str, container_path: str):
         else:
             cloud_suffix = _get_storage_endpoint_from_metadata()
-        self._account_uri = f"https://{storage_name}.blob.{cloud_suffix}"
-
-        # Its possible that the account URL may need additional tweaking to add a SAS
-        # token if the account does not allow for anonymous access. However, for
-        # performance reasons take a lazy approach to determining this and wait
-        # until a client of this class actually attempts to reference the URI
-        super().__init__(
-            PathType.AZUREBLOB,
-            None,
-        )
+        return f"https://{storage_name}.blob.{cloud_suffix}"
 
-    @property
-    def uri(self) -> str:
-        """Asset URI."""
-        # If we have already cached the URI, then simply return it
-        if self._uri is not None:
-            return self._uri
-
-        # Build the simple, non-SAS URI for quick use later in the function.
-        # Its possible that the account URI may need additional tweaking to add a SAS
-        # token if the account does not allow for anonymous access.
-        self._uri = f"{self._account_uri}/{self._container_name}/{self._container_path}"
+    @staticmethod
+    def generate_uri(uri: str, account_uri: str, container_name: str, token: str) -> str:
+        """Generate URI.
 
+        :param uri: Asset URI
+        :type uri: str
+        :param account_uri: Account URI
+        :type account_uri: str
+        :param container_name: Blob container name
+        :type container_name: str
+        :param token: SAS token
+        :type token: str
+        """
         # If a SAS token has been explicitly set, then assume that the URI+token is valid.
         # Simply append the token and return. If no SAS token is set, then proceed to rest of function.
-        if self.token is not None:
-            self._uri += "?" + self.token
-            return self._uri
+        if token is not None:
+            uri += "?" + token
+            return uri
 
         # The first test is to see if we can simply list the contents of the container
         # using a very simple and quick HTTP request. In order for this to work,
         # container must support anonymous read access. If this succeeds, we can
         # return the URI "as-is".
         try:
-            list_container_url = f"{self._account_uri}/{self._container_name}?restype=container&comp=list&maxresults=1"
+            list_container_url = f"{account_uri}/{container_name}?restype=container&comp=list&maxresults=1"
             response = requests.get(
                 list_container_url,
                 timeout=AzureBlobstoreAssetPath.CONTAINER_ANONYMOUS_ACCESS_CHECK_TIMEOUT
             )
 
             if response.status_code >= 200 and response.status_code <= 299:
-                return self._uri
+                return uri
         except Exception:
             # If we fail pass through to the next approach
-            pass
+            return None

```

Comment:
Originally this function would add the SAS token if the test above demonstrated that the container isn't public. I think you'll need to generate and add the token at this point.